crates/tests/signer-tests/sign_blinded_message.rs

```diff
@@ -0,0 +1,193 @@
+        })
+        .await?
+        .into_inner()
+        .signatures;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn invalid_secret() -> Result<()> {
+    let mut client = init_signer_client().await?;
+
+    let res = client
+        .declare_keyset(DeclareKeysetRequest {
+            unit: "strk".to_string(),
+            index: 1,
+            max_order: 32,
+        })
+        .await?;
+
+    let declare_keyset_response: DeclareKeysetResponse = res.into_inner();
+
+    let keyset_id = KeysetId::from_iter(
+        declare_keyset_response
+            .clone()
+            .keys
+            .into_iter()
+            .map(|k| PublicKey::from_str(&k.pubkey).unwrap()),
+    );
+    let public_keys: Vec<String> = declare_keyset_response
+        .keys
+        .iter()
+        .map(|k| k.pubkey.clone())
+        .collect();
+
+    let public_key =
+        PublicKey::from_str("031d032d4042b010310f9ed54b12c819afc334505752e9e31a9fb13a43ad2eea2");
+
+    assert!(public_key.is_err());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn amount_not_present() -> Result<()> {
+    let mut client = init_signer_client().await?;
+
+    let res = client
+        .declare_keyset(DeclareKeysetRequest {
+            unit: "strk".to_string(),
+            index: 1,
+            max_order: 32,
+        })
+        .await?;
+
+    let declare_keyset_response: DeclareKeysetResponse = res.into_inner();
+
+    let keyset_id = KeysetId::from_iter(
+        declare_keyset_response
+            .clone()
+            .keys
+            .into_iter()
+            .map(|k| PublicKey::from_str(&k.pubkey).unwrap()),
+    );
+    let public_keys: Vec<String> = declare_keyset_response
+        .keys
+        .iter()
+        .map(|k| k.pubkey.clone())
+        .collect();
+
+    let public_key = PublicKey::from_str(&public_keys[0]).unwrap();
+
+    let blinded_message = BlindedMessage {
+        amount: Amount::from_i64_repr(7),
+        keyset_id,
+        blinded_secret: public_key,
+    };
+
+    let blind_signatures = client
+        .sign_blinded_messages(SignBlindedMessagesRequest {
+            messages: [blinded_message]
+                .iter()
+                .map(|bm| signer::BlindedMessage {
+                    amount: bm.amount.into(),
+                    keyset_id: bm.keyset_id.to_bytes().to_vec(),
+                    blinded_secret: bm.blinded_secret.to_bytes().to_vec(),
+                })
+                .collect(),
+        })
+        .await;
+
+    assert!(blind_signatures.is_err());
+    Ok(())
+}
+
+#[tokio::test]
+async fn non_existent_keysetid() -> Result<()> {
+    let mut client = init_signer_client().await?;
+
+    let res = client
+        .declare_keyset(DeclareKeysetRequest {
+            unit: "strk".to_string(),
+            index: 1,
+            max_order: 32,
+        })
+        .await?;
+
+    let keyset_id =
+        KeysetId::from_str("034a013c42af6d4ae91e67de936d6039e0e35899d0c3936a59c9292b9646adf68");
+
+    assert!(keyset_id.is_err());
```

Comment:
no. we want you to send a SignBlindedMessages request, with string mapping to a non existing keyset id. and get the error from the signer service